concordium-consensus/testdata/contracts/v1/transfer-cases.wat

```diff
@@ -0,0 +1,38 @@
+(module
+
+  ;; Imports
+  (import "concordium" "get_parameter_section" (func $host_get_parameter_section (param $index i32) (param $write_location i32) (param $length i32) (param $offset i32) (result i32)))
+
+  (import "concordium" "invoke" (func $host_invoke (param $tag i32) (param $start i32) (param $length i32) (result i64)))
+  (import "concordium" "state_create_entry" (func $state_create_entry (param $key_start i32) (param $key_length i32) (result i64)))
+  (import "concordium" "state_entry_write" (func $state_entry_write (param $entry i64) (param $read_location i32) (param $length i32) (param $offset i32) (result i32)))
+
+
+  ;; Initialize contract.
+  (func $init (export "init_contract") (param i64) (result i32)
+        (return (i32.const 0))) ;; Successful init
+
+  (func $receive_upgrade (export "contract.transfer") (param $amount i64) (result i32)
+       ;; Read the module reference from the parameter into memory, and on byte 33 the test scenario
```

Comment:
I don't think this reflects the actual code.
```suggestion
       ;; Read account address (32 bytes), an amount (8 bytes) used for transfer and a flag (1 byte) from the parameter into memory. Setting the flag to: 0 means do not update the state, 1 means update the state before invoking transfer, 2 means update the state after invoking transfer.
```

Reply:
Copy paste :)